pkg/controller/plan/kubevirt.go

```diff
@@ -1767,8 +1767,8 @@ func (r *KubeVirt) createOpenStackVolumes(vm ref.Ref) (pvcNames []string, err er
 
 	storageName := r.Context.Map.Storage.Spec.Map[0].Destination.StorageClass
 
-	if len(openstackVm.AttachedVolumes) > 0 {
-		for _, vol := range openstackVm.AttachedVolumes {
+	if len(openstackVm.Volumes) > 0 {
```

Comment:
does it mean that we'll transfer volumes that are not attached to the vm and attach them to the imported vm?

Reply:
No, the `Workload` field was renamed from `AttachedVolumes` to `Volumes` but it's the same thing.
This is because the `VM` definition already has an `AttachedVolumes` field which only contains the `IDs` of the volumes attached to it and having `Workload.XVM.AttachedVolumes`  and `Workload.AttachedVolumes` (containing all the attached volume details) is confusing:
```go
// Workload
type Workload struct {
	SelfLink string `json:"selfLink"`
	XVM
}

// Expanded: VM.
type XVM struct {
	VM
	Image       Image        `json:"image"`
	Flavor      Flavor       `json:"flavor"`
	Networks    []Network    `json:"networks"`
	Subnets     []Subnet     `json:"subnets"`
	Volumes     []Volume     `json:"volumes,omitempty"`
	VolumeTypes []VolumeType `json:"volumeTypes,omitempty"`
	Snapshots   []Snapshot   `json:"snapshots,omitempty"`
}
```

So  this way `Workload.AttachedVolumes` contains the attached volumes IDs and ` `Workload.Volumes` contain all the details of those attached volumes.